docs/src/content/hardhat-runner/docs/guides/test-contracts.md

```diff
@@ -1,3 +1,215 @@
+
+Finally, we check that the value returned by the `unlockTime()` getter in the contract matches the value that we used when we deployed it. Since all the functions in a contract are async, we have to use the `await` keyword to get its value, otherwise we would be comparing a promise with a number and this would always fail.
+
+### Testing a function that reverts
+
+In the previous test we checked that a getter function returned the correct value. This was a read-only function that can be called without paying a fee and without any risk. Other functions, however, can modify the state of the contract, like the `withdraw` function in the `Lock` contract. This means that we want some pre-conditions to be met for this function to be called successfully. If you look at its first lines you’ll see a couple of `require` checks for that purpose:
+
+```solidity
+function withdraw() public {
+  require(block.timestamp >= unlockTime, "You can't withdraw yet");
+  require(msg.sender == owner, "You aren't the owner");
+```
+
+The first statement checks that the unlock time has been reached, and the second one checks that the address calling the contract is its owner. Let’s start by writing a test for the first pre-condition:
+
+```tsx
+it("Should revert with the right error if called too soon", async function () {
+  // ...deploy the contract as before...
+
+  await expect(lock.withdraw()).to.be.revertedWith("You can't withdraw yet");
+});
+```
+
+In the previous test we used `.to.equal`, which is part of Chai and it’s used to compare two values, but here we are using [`.to.be.revertedWith`](/hardhat-chai-matchers/docs/reference#.revertedwith). This matcher asserts that a transaction reverts, and that the reason string of the revert is equal to a given string. The `.to.be.revertedWith` matcher is not part of Chai itself; instead, it’s added by the [Hardhat Chai Matchers](/hardhat-chai-matchers) plugin, which is included in the sample project we are using.
+
+Notice that in the previous test we wrote `expect(await ...)` but now we are doing `await expect(...)`. In the first case we were comparing two values in a synchronous way; the inner await is just there to wait for the value to be retrieved. In the second case, the whole assertion is async, because it has to wait until the transaction is mined. This means that the `expect` call returns a promise that we have to wait.
```

Comment:
```suggestion
Notice that in the previous test we wrote `expect(await ...)` but now we are doing `await expect(...)`. In the first case we were comparing two values in a synchronous way; the inner await is just there to wait for the value to be retrieved. In the second case, the whole assertion is async because it has to wait until the transaction is mined. This means that the `expect` call returns a promise that we have to await.
```